lang/python/airr/interface.py

```diff
@@ -196,3 +200,148 @@ def validate_rearrangement(filename, debug=False):
+
+    Returns:
+      dictionary of repertoire objects.
+    """
+
+    # Because the repertoires are read in completely, we do not bother
+    # with a reader class.
+    md = None
+
+    # determine file type from extension and use appropriate loader
+    ext = filename.split('.')[-1]
+    if ext == 'yaml' or ext == 'yml':
+        with open(filename, 'r', encoding='utf-8') as handle:
+            md = yaml.load(handle, Loader=yamlordereddictloader.Loader)
+    elif ext == 'json':
+        with open(filename, 'r', encoding='utf-8') as handle:
+            md = json.load(handle)
+    else:
+        if debug:
+            sys.stderr.write('Unknown file type: %s. Supported file extensions are "yaml", "yml" or "json"\n' % (ext))
+        raise TypeError('Unknown file type: %s. Supported file extensions are "yaml", "yml" or "json"\n' % (ext))
+
+    if md.get('Repertoire') is None:
+        if debug:
+            sys.stderr.write('%s is missing "Repertoire" key\n' % (filename))
+        raise KeyError('Repertoire object cannot be found in the file')
+
+    # validate if requested
+
+    # we do not perform any additional processing
+    return md
+
+
+def validate_repertoire(filename, debug=False):
+    """
+    Validates an AIRR repertoire metadata file
+
+    Arguments:
+      filename (str): path of the file to validate.
+      debug (bool): debug flag. If True print debugging information to standard error.
+
+    Returns:
+      bool: True if files passed validation, otherwise False.
+    """
+    valid = True
+    if debug:
+        sys.stderr.write('Validating: %s\n' % filename)
+
+    # load with validate
+    try:
+        data = load_repertoire(filename, validate=True, debug=debug)
+    except TypeError:
+        valid = False
+    except KeyError:
+        valid = False
+    except ValidationError as e:
+        valid = False
+        if debug:
+            sys.stderr.write('%s has validation error: %s\n' % (filename, e))
+
+    # Validate each repertoire
+    reps = data['Repertoire']
+    i = 0
+    for r in reps:
+        try:
+            RepertoireSchema.validate_object(r)
+            i = i + 1
+        except ValidationError as e:
+            valid = False
+            if debug:
+                sys.stderr.write('%s has repertoire at array position %i with validation error: %s\n' % (filename, i, e))
+
+    return valid
+
+def write_repertoire(filename, repertoires, info=None, debug=False):
+    """
+    Write an AIRR repertoire metadata file
+
+    Arguments:
+      file (str): path to the output file.
+      repertoires (list): array of repertoire objects.
+      info (object): info object to write. Will write current AIRR Schema info if not specified.
+      debug (bool): debug flag. If True print debugging information to standard error.
+
+    Returns:
+      bool: True if the file is written without error.
+    """
+    if not isinstance(repertoires, list):
```

Comment:
No objection to requiring it be a list, but does it need to be? Could it be any iterable instead (which would also include a list)? Eg,

```
try:
    iter(repertoires)
except TypeError:
    if debug:  sys.stderr.write('Repertoires parameter is not iterable\n')
```

Reply:
I'm using `yaml.dump` and `json.dump` to output the whole object. I suppose if it let's me assign the iterator to the metadata `md` object and it outputs to the file then I'm okay. If I actually have to iterate over the object to pull out the data, then it's less useful.

```
    md['Repertoire'] = repertoires
```

Reply:
Ah, I think I missed that the list was a single object (ie, `yaml.dump` instead of `yaml.dump_all`). Nevermind.